### PR TITLE
`cider-connect-clj&cljs`: don't render `"ClojureScript REPL type:" for JVM repls

### PR DESCRIPTION
**Pull request by vemv**
> Fixes https://github.com/clojure-emacs/cider/issues/3383

I verified this one locally.

Cheers - V

### CODE REVIEW

File: cider.el [hunk 1]
```diff
@@ -1448,10 +1448,13 @@ non-nil, don't start if ClojureScript requirements are not met."
...
-         (clj-repl (cider-connect-clj params)))
-    (if soft-cljs-start
-        (when (cider--check-cljs (plist-get params :cljs-repl-type) 'no-error)
-          (cider-connect-sibling-cljs params clj-repl))
+         (clj-params (thread-first
+                       params
+                       copy-sequence
+                       (map-delete :cljs-repl-type)))
+         (clj-repl (cider-connect-clj clj-params)))
+    (when (or (not soft-cljs-start)
```
Comment (vemv) [inline cider.el:1456]: Also simplified this condition, since `(cider-connect-sibling-cljs params clj-repl)` was duplicated.

Comment (bbatsov) [reply cider.el:1456]: Not a big fan of this negation in the condition, though.

Comment (vemv) [reply cider.el:1456]: Maybe?

```lisp
(if soft-cljs-start
    (cider--check-cljs (plist-get params :cljs-repl-type) 'no-error)
  t)
```

Comment (bbatsov) [reply cider.el:1456]: Much better IMO.

**Review by bbatsov — APPROVED** [review]

**Pull request merged**